### PR TITLE
Metrics verdict

### PR DESCRIPTION
**Pull request by Elfoniok**
Verification of images based on SSIM and image correlation.

### CODE REVIEW

File: apps/blender/task/blendercropper.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        for _ in range(splits_num):
+            split_x = self._random_split(left, right, self.crop_size[0])
+            split_y = self._random_split(top, bottom, self.crop_size[1])
+            split_values.append((split_x[0], split_x[1], split_y[0],
+                                 split_y[1]))
+            split_pixels.append(self._pixel(resolution, split_x[0], split_y[1],
+                                            left, bottom))
+        return split_values, split_pixels
+
+    # pylint: disable-msg=too-many-arguments
```
Comment (jiivan) [inline apps/blender/task/blendercropper.py]: This pylint directive should better be placed on a line that it's needed to narrow it's scope. And truly speaking there sure are lot of arguments...

File: apps/blender/task/blendercropper.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    def render_crops(self, computer, resources, crop_rendered,
+                     crop_render_failure, subtask_info,
+                     num_crops=3,
+                     crop_size=None):
+        # pylint: disable=unused-argument
```
Comment (jiivan) [inline apps/blender/task/blendercropper.py:98]: ditto

**Review by jiivan — COMMENTED** [review]
LGTM, I only don't understand the linter disabling parts.

File: apps/blender/task/blendercropper.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import os
+from copy import deepcopy
+from functools import partial
+import numpy as np
+
+from apps.blender.resources.scenefileeditor import generate_blender_crop_file
+from golem.core.common import timeout_to_deadline
+
+
+# pylint: disable=R0903
```
Comment (jiivan) [inline apps/blender/task/blendercropper.py]: This is very reasonable warning from pylint, and why disable it for a whole file?

http://pylint-messages.wikidot.com/messages:r0903

File: apps/blender/task/blendercropper.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                                  crops_info[0][self.crop_counter],
+                                  subtask_info, verify_ctx, crop_rendered,
+                                  crop_render_failure)
+            self.crop_counter += 1
+        return self.crop_size
+
+    # FIXME it would be better to make this subtask agnostic, pass only data
+    # needed to generate crops. Drop local computer.
+    # pylint: disable-msg=too-many-arguments
+    # pylint: disable=R0914
```
Comment (jiivan) [inline apps/blender/task/blendercropper.py:121]: Looks like 914 is spot on readability here...

http://pylint-messages.wikidot.com/messages:r0914

File: apps/blender/task/verifier.py [hunk 5]
```diff
@@ -1,40 +1,25 @@
...
-# pylint: disable=R0903
-class VerificationContext:
-    def __init__(self, crops_position, crop_id, crops_path):
-        self.crop_id = crop_id
-        self.crop_path = os.path.join(crops_path, str(crop_id))
-        self.crop_position_x = crops_position[crop_id][0]
-        self.crop_position_y = crops_position[crop_id][1]
-
 
 # pylint: disable=R0902
```
Comment (jiivan) [inline apps/blender/task/verifier.py]: There sure is a lint here.

http://pylint-messages.wikidot.com/messages:r0903

File: apps/blender/task/verifier.py [hunk 6]
```diff
@@ -110,54 +103,7 @@ def success():
...
-        ctd['extra_data']['outfilebasename'] = \
-            "ref_" + subtask_info['outfilebasename']
-        ctd['extra_data']['script_src'] = script_src
-        ctd['deadline'] = timeout_to_deadline(subtask_info['subtask_timeout'])
-        return ctd
-
-    #  The verification function will generate three random crops, from results
+    # The verification function will generate three random crops, from results
     #  only after all three will be generated, we can start verification process
     # pylint: disable=R0914
```
Comment (jiivan) [inline apps/blender/task/verifier.py:114]: http://pylint-messages.wikidot.com/messages:r0914

File: tests/apps/blender/task/test_blender_verification.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+from golem.testutils import TempDirFixture
+from golem.docker.image import DockerImage
+from golem.task.localcomputer import ComputerAdapter
+from golem.tools.ci import ci_skip
+
+
+class TestGenerateCrops(TempDirFixture):
+
+    def setUp(self):
+        # pylint: disable=R0915
```
Comment (jiivan) [inline tests/apps/blender/task/test_blender_verification.py:17]: R0915: Too many statements

File: tests/apps/blender/task/test_blender_verification.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+        self.subtask_info['ctd']['task_owner'] = dict()
+        self.subtask_info['ctd']['working_directory'] = self.tempdir
+
+    @ci_skip
+    def test_bad_image(self):
+
+        d = Deferred()
+
+        def success(*args, **kwargs):
+            # pylint: disable=unused-argument
```
Comment (jiivan) [inline tests/apps/blender/task/test_blender_verification.py:93]: 👍

**Review by jiivan — DISMISSED** [review]
I know some of this code doesn't come from you, but It's hard to justify all those linter switch offs.

**Review by mfranciszkiewicz — APPROVED** [review]

**Review by badb — APPROVED** [review]

**Pull request merged**